### PR TITLE
Added initial security considerations.

### PR DESCRIPTION
**Pull request by birenroy**
First pass at the section on security considerations.

### CODE REVIEW

**Review by diannahu — APPROVED** [review]
Looks great!

File: draft-beky-httpbis-metadata.md [hunk 1]
```diff
@@ -128,8 +128,9 @@ METADATA frames obey the maximum frame size set by SETTINGS_MAX_FRAME_SIZE.
 METADATA frames are not subject to flow control.
 
 The metadata block of an HTTP/2 METADATA frame is encoded using HPACK
-instructions ({{!RFC7541}}).  An endpoint _MUST NOT_ use any HPACK instructions
-that change the dynamic table.
+representations ({{!RFC7541}}).  An endpoint _MUST NOT_ use any HPACK representations
+that change the dynamic table; any METADATA frame with such representations _SHOULD_
```
Comment (DavidSchinazi) [inline draft-beky-httpbis-metadata.md:132]:
```suggestion
that change the dynamic table inside METADATA frames; any METADATA frame with such representations _SHOULD_
```

File: draft-beky-httpbis-metadata.md [hunk 2]
```diff
@@ -152,7 +153,8 @@ with the exchange carried by that stream.
 
 The metadata block of a HTTP/3 METADATA frame is encoded using QPACK
 representations.  An endpoint _MUST NOT_ use any QPACK representations that
-reference the dynamic table.  Therefore the Required Insert Count is be zero,
+reference the dynamic table; any METADATA frame with such representations _SHOULD_
```
Comment (DavidSchinazi) [inline draft-beky-httpbis-metadata.md:156]:
```suggestion
reference the dynamic table inside METADATA frames; any METADATA frame with such representations _SHOULD_
```

File: draft-beky-httpbis-metadata.md [hunk 3]
```diff
@@ -176,8 +178,22 @@ NOT_ send METADATA frames after it learns that the peer does not support them.
...
+possibility of changes to the compression state of a connection.  However,
+METADATA frames are extension frames, and might be dropped by implementations or
+intermediaries.  To avoid the problem of compression state desynchronization
+between endpoints, HPACK or QPACK representations that change compression state
+are disallowed.
+
+## Denial-of-Service Considerations
+
+Depending on the application, metadata blocks sent over HTTP/2 might be larger
+than the negotiated SETTINGS_MAX_FRAME_SIZE.  Although RFC 9113 does not
```
[the remark this answers is not in the document]
Comment (bencebeky) [reply draft-beky-httpbis-metadata.md:193]: This is a good point.  However, RFC 9113 does not explicitly disallow extensions from modifying the protocol is such a way that new frames are exempt from SETTINGS_MAX_FRAME_SIZE, neither in Section 4.2 nor in Section 5.5.  One might interpret Section 5.5 as "all bets are off" regarding extensions as long as interoperability with existing compliant implementations is not compromised.  So there might be some room for interpretation here.

I think it is worth making it explicit here that METADATA frames are subject to SETTINGS_MAX_FRAME_SIZE.  However, I suggest softening up the language and not making a statement about what RFC 9113 states in this regard.

**Pull request merged**